mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+            search_all_experiments=True,
+            filter_string=f"attributes.run_id = '{pointer}'",
+            output_format="list",
+        )
+        if not run:
+            raise ValueError(
+                "The provided run id was not found in the set MLFlow registry. Try to set the registry using "
+                "MLFlow's environment variables."
+            )
+        run = run[0]  # We are using a run id, so only one will be returned in the list.
+        # Create a run spec and metadata for creating the run object to hold the mlflow run:
+        run_spec = RunSpec(function=function_name, handler=handler)
+        run_metadata = RunMetadata(
+            uid=run.info.run_uuid, name=run.info.run_name, project=project.name
+        )
+        run_object = RunObject(spec=run_spec, metadata=run_metadata)
+        # Create a context from the run object:
+        ctx = mlrun.get_or_create_ctx(
+            name=run.info.run_name,
+            spec=run_object,
+        )
+        # Store the run in the MLRun DB, then import the MLFlow data to it:
+        ctx.store_run()
+        cls._log_run(context=ctx, run=run, is_offline=True)
+
+        # Create a rundb in order to update the run's state as completed (can't be done using context)
+        rundb = mlrun.get_run_db()
+        project = run_object.metadata.project
+        uid = run_object.metadata.uid
+        updates = {
+            "status.last_update": now_date().isoformat(),
+            "status.state": "completed",
+        }
+        run_object.status.state = "completed"
+        rundb.update_run(updates, uid, project)
+
+        # Print a summary message after importing the run:
+        # RuntimeMock needed for mocking runtime parameter in `_log_track_results`
+        class RuntimeMock:
+            is_child = False
+
+        result = ctx.to_dict()
+        run_object = RunObject.from_dict(result)
+        ClientBaseLauncher._log_track_results(
+            runtime=RuntimeMock, result=result, run=run_object
+        )
+
+        return run_object
+
+    @classmethod
+    def import_model(
+        cls,
+        project: MlrunProject,
+        pointer: str,
+        key: str = None,
+        metrics: dict = None,
+        extra_data: dict = None,
+    ) -> ModelArtifact:
+        """
+        Import a model from MLFlow to MLRun.
+
+        :param project:    The MLRun project to import the model to.
+        :param pointer:    The MLFlow model uri to import.
+        :param key:        The model key to be used in MLRun. Mandatory for importing.
+        :param metrics:    The model's metrics.
+        :param extra_data: Extra artifacts and files to log with the model.
+
+        :return: The newly imported ModelArtifact.
+        """
+        # Validate key is given:
+        if key is None:
+            raise ValueError(
+                "MLFlow models require a key to import into MLRun - the key of the model artifact to be created."
+            )
+
```

Comment:
so maybe key should not default to None?

Reply:
agreed

Reply:
same reason as before, it is an inheriting function. In this case, I think we should not change it because in some other vendors we might have a key to get from the object itself.